kmod/patch/Makefile

```diff
@@ -18,12 +18,15 @@ $(KPATCH_NAME)-objs += patch-hook.o output.o
 
 all: $(KPATCH_NAME).ko
 
-$(KPATCH_NAME).ko:
+build.env:
+	export -p > $@
+
+$(KPATCH_NAME).ko: build.env
 	$(KPATCH_MAKE)
 
 patch-hook.o: patch-hook.c kpatch-patch-hook.c livepatch-patch-hook.c
 	$(KPATCH_MAKE) patch-hook.o
 
 clean:
-	$(RM) -Rf .*.o.cmd .*.ko.cmd .tmp_versions *.o *.ko *.mod.c \
+	$(RM) -Rf .*.o.cmd .*.ko.cmd .tmp_versions *[^output].o *.ko *.mod.c \
```

Comment:
This doesn't have intended result. Any .o file containing o, u, t, p in it's name will be skipped. I am afraid we'll have to use make magic. This should work: `$(filter-out output.o, $(wildcard *.o))`.

Reply:
Whoops, yeah you're right about that.  Make magic it is then.